parquet-common/src/main/java/org/apache/parquet/bytes/CapacityByteArrayOutputStream.java

```diff
@@ -205,6 +206,12 @@ public void write(byte b[], int off, int len) {
       throw new IndexOutOfBoundsException(
           String.format("Given byte array of size %d, with requested length(%d) and offset(%d)", b.length, len, off));
     }
+    int targetBytesUsed = bytesUsed + len;
```

Comment:
Thanks for the change!

What about using `bytesUsed = Math.addExact(bytesUsed, len)`. It is available in java8 and looks more elegant to throw ArithmeticException by itself.

https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Math.html#addExact(int,int)